### PR TITLE
[Fix]: Fix CreateUser query 

### PR DESCRIPTION
**Pull request by oreoluwade**
#### What does this PR do?
- Adjusts the possible query options on the `createUser` mutation due to changes on the backend

#### Description of Task to be completed?
- Adjust query options on the `createUser` mutation.
- Adjust query options in the mock client
- Update select apollo dependencies, refactor client based on upgrade
- Update snapshots

#### How should this be manually tested?
- Check out into the `feature/fix-create-user-query-options`.
- Run `yarn` or `yarn install`
- Run the app with `yarn start`
- It should work as before with no errors on the browser console.

#### What are the relevant pivotal tracker stories?
[#158291479](https://www.pivotaltracker.com/story/show/158291479)

### CODE REVIEW

File: src/apollo-client/apollo-client.js [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
-      authorization: localStorage.getItem('token') || null,
-    },
+      authorization: token || null
+    }
   });
-  return forward(operation);
-});
+};
+
+const requestLink = new ApolloLink((operation, forward) =>
```
Comment (andela-mabdussalam) [inline src/apollo-client/apollo-client.js]: @oreoluwade Pls could you help me understand what this request link is about? Thanks.

Comment (oreoluwade) [reply src/apollo-client/apollo-client.js]: @omorilewa, first of all, we need to revoke commenting access to @andela-mabdussalam.

TLDR: Requests need the token before they can be carried out.

Secondly, the requestLink is the middleware link each request to the server passes through. The token has to be available for those requests to be "parsed" (permit my usage of the word). The drama that seems to be happening in all those lines is the standard configuration object I copped from [here](https://www.apollographql.com/docs/react/advanced/boost-migration.html). Will read more about observables subsequently.

**Comment by oreoluwade** [comment]
@omorilewa Please, if you are requesting changes, you need to be specific about what changes you are requesting. I would not go ahead to merge if you have questions (Cos I imagine your request was a bid to block the PR from being merged. Unless I am wrong).

File: src/apollo-client/apollo-client.js (hunk 1 above)
Comment (oreoluwade) [reply src/apollo-client/apollo-client.js]: also @omorilewa, the `requestLink` and `request` drama up there are a variant of what was the middleware link before. This seems to be the latest way to configure it is why I changed it. 

Meanwhile, I hold my hands up in surrender that these changes do not reflect what this PR is supposed to be about at its core. For that, I ask for forgiveness.

**Comment by omorilewa** [comment]
@oreoluwade I could not assume that you would not go ahead to merge, hence my blocking the merge. Thanks for referring me to the documentation, I had checked it earlier. You can ignore my question, I thought there may be additional context you could have provided.

**Comment by oreoluwade** [comment]
@omorilewa I am sorry if in some way, I have put myself forward as dishonest or dubious in the past, going ahead to do stuff without considering your valuable input. I promise that you can trust me henceforth on that; no hurried merging `especially` when you or anyone has concerns or questions.
